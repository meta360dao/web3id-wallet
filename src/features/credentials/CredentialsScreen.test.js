import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {CredentialsScreen, CredentialListItem} from './CredentialsScreen';

const mockStore = configureMockStore();

describe('CredentialsScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      <CredentialsScreen credentials={[]} onRemove={jest.fn} onAdd={jest.fn} />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('should render correctly credential list', () => {
    const initialState = {
      // placeholder for redux store
    };

    const credential = {
      '@context': [
        'https://www.w3.org/2018/credentials/v1',
        {
          dk: 'https://ld.dock.io/credentials#',
          EmployeeIDCredential: 'dk:EmployeeIDCredential',
          name: 'dk:name',
          email: 'dk:email',
          employeeId: 'dk:employeeId',
        },
      ],
      id: 'https://creds-testnet.dock.io/191ce57b4b9195d823937c016e31544c225279e4fb3e7d5ed88038acf5daac42',
      type: ['VerifiableCredential', 'EmployeeIDCredential'],
      credentialSubject: {
        name: 'Alice Doe',
        email: 'alice@dock.io',
        employeeId: 123456,
      },
      issuanceDate: '2022-08-29T15:24:47.799Z',
      name: 'Acme Employee ID',
      proof: {
        type: 'Ed25519Signature2018',
        created: '2022-08-29T16:24:47Z',
        verificationMethod:
          'did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA#keys-1',
        proofPurpose: 'assertionMethod',
        proofValue:
          'zYetMX6EPE1Lzqe92ZZopqFQrwN1g9wwXE2UHW6y69cDZQfoCkz7hMZDxURvo8fVW97p2f6tpAfWKEQU8RvRNPxC',
      },
      issuer: 'did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA',
    };
    const formattedData = {
      humanizedType: 'Employee I D Credential',
      title: 'Acme Employee ID',
      subjectName: 'Alice Doe',
      issuerName: 'did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA',
      date: 'August 29, 2022',
      image: null,
      images: {
        issuerImage: null,
        subjectImage: null,
        mainImage: null,
        imagesList: [],
      },
      documents: [[]],
      template: 'credential',
      qrImage: false,
      attributes: [
        {name: 'Name', property: 'name', value: 'Alice Doe'},
        {name: 'Email', property: 'email', value: 'alice@dock.io'},
        {name: 'Employee Id', property: 'employeeId', value: 123456},
      ],
      subjects: [
        {name: 'Alice Doe', email: 'alice@dock.io', employeeId: 123456},
      ],
      issuer: 'did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA',
      issuanceDate: '2022-08-29T15:24:47.799Z',
      expirationDateStr: '',
      dateStr: 'August 29, 2022',
    };
    const wrapper = shallow(
      <CredentialListItem
        credential={credential}
        formattedData={formattedData}
        onRemove={jest.fn}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
