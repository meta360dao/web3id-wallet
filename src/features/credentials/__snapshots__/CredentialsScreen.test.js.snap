// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CredentialsScreen should render correctly 1`] = `
<RNCSafeAreaView
  accessibilityLabel="CredentialsScreen"
  style={
    Object {
      "backgroundColor": "#000",
      "flex": 1,
    }
  }
  testID="CredentialsScreen"
>
  <AppGlobalHeader />
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(View)>
        <Box
          alignItems="center"
          flexDirection="row"
          marginLeft={22}
          marginRight={22}
        >
          <Box
            flex={1}
          >
            <Styled(Component)
              fontFamily="Montserrat"
              fontSize={24}
              fontWeight="600"
            >
              Credentials
            </Styled(Component)>
          </Box>
          <Box
            row={true}
          >
            <IconButton
              col={true}
              onPress={[Function]}
            >
              <SvgFile />
            </IconButton>
          </Box>
        </Box>
      </Styled(View)>
      <Styled(ScrollView)>
        <EmptyCredentials
          mt="50%"
        />
      </Styled(ScrollView)>
    </Memo(ForwardRef(Box))>
    <Memo(ForwardRef(Box))
      flex={1}
      flexGrow={0.11}
    >
      <TabNavigation />
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;

exports[`CredentialsScreen should render correctly credential list 1`] = `
<Component
  credential={
    Object {
      "@context": Array [
        "https://www.w3.org/2018/credentials/v1",
        Object {
          "EmployeeIDCredential": "dk:EmployeeIDCredential",
          "dk": "https://ld.dock.io/credentials#",
          "email": "dk:email",
          "employeeId": "dk:employeeId",
          "name": "dk:name",
        },
      ],
      "credentialSubject": Object {
        "email": "alice@dock.io",
        "employeeId": 123456,
        "name": "Alice Doe",
      },
      "id": "https://creds-testnet.dock.io/191ce57b4b9195d823937c016e31544c225279e4fb3e7d5ed88038acf5daac42",
      "issuanceDate": "2022-08-29T15:24:47.799Z",
      "issuer": "did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA",
      "name": "Acme Employee ID",
      "proof": Object {
        "created": "2022-08-29T16:24:47Z",
        "proofPurpose": "assertionMethod",
        "proofValue": "zYetMX6EPE1Lzqe92ZZopqFQrwN1g9wwXE2UHW6y69cDZQfoCkz7hMZDxURvo8fVW97p2f6tpAfWKEQU8RvRNPxC",
        "type": "Ed25519Signature2018",
        "verificationMethod": "did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA#keys-1",
      },
      "type": Array [
        "VerifiableCredential",
        "EmployeeIDCredential",
      ],
    }
  }
  formattedData={
    Object {
      "attributes": Array [
        Object {
          "name": "Name",
          "property": "name",
          "value": "Alice Doe",
        },
        Object {
          "name": "Email",
          "property": "email",
          "value": "alice@dock.io",
        },
        Object {
          "name": "Employee Id",
          "property": "employeeId",
          "value": 123456,
        },
      ],
      "date": "August 29, 2022",
      "dateStr": "August 29, 2022",
      "documents": Array [
        Array [],
      ],
      "expirationDateStr": "",
      "humanizedType": "Employee I D Credential",
      "image": null,
      "images": Object {
        "imagesList": Array [],
        "issuerImage": null,
        "mainImage": null,
        "subjectImage": null,
      },
      "issuanceDate": "2022-08-29T15:24:47.799Z",
      "issuer": "did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA",
      "issuerName": "did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA",
      "qrImage": false,
      "subjectName": "Alice Doe",
      "subjects": Array [
        Object {
          "email": "alice@dock.io",
          "employeeId": 123456,
          "name": "Alice Doe",
        },
      ],
      "template": "credential",
      "title": "Acme Employee ID",
    }
  }
  onRemove={[Function]}
/>
`;
